examples/neighbors/plot_lmnn_illustration.py

```diff
@@ -0,0 +1,153 @@
+"""
+==========================================
+Large Margin Nearest Neighbor Illustration
+==========================================
+
+An example illustrating the goal of learning a distance metric that maximizes
+the nearest neighbors classification accuracy. The example is solely for
+illustration purposes. Please refer to the :ref:`User Guide <neighbors>` for
+more information.
+"""
+
+# Author: John Chiotellis <johnyc.code@gmail.com>
+# License: BSD 3 clause
+
+import numpy as np
+import matplotlib.pyplot as plt
+from sklearn.datasets import make_classification
+from sklearn.neighbors import LargeMarginNearestNeighbor
+from sklearn.neighbors.lmnn import _select_target_neighbors
```

Comment:
I don't think you should be using a private function in an example. is there merit to making it public?

Reply:
I had it as public before, but @agramfort suggested 
> keeping it private for now and exposing it if there is a clear need is easier than the other way around

Reply:
Maybe he'd not realised it was used in an example.

Reply:
Yes, it may have been before I pushed the example.